packages/headless/src/utils/validate-payload.test.ts

```diff
@@ -42,6 +42,16 @@ describe('validatePayload', () => {
     expect(validatedPayload.payload).toEqual(11);
     expect(validatedPayload.error).toBeDefined();
   });
+
+  it('when SchemaDefinition is an object, validating a payload that is not an object should be invalid', () => {
+    const notAnObject = ['foo', 11, true];
+    for (const testCase in notAnObject) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Have to use this here, since we actually want to test what happens when someone send in `any`. 

Remember that this fix is ultimately for JavaScript users. 

TypeScript users would not normally be able to pass in "anything" in an action (well... unless they cast to `any` :P )